_applications/my-first-radanalytics-app.adoc

```diff
@@ -0,0 +1,113 @@
+
+Speaking of https://docs.openshift.org/latest/architecture/core_concepts/builds_and_image_streams.html#source-build[source-to-image], this technology is fundamental to the operation of this tutorial. Source-to-image provides a convenient way for you to produce container images directly from your source repositories. This work is handled by language specific builder images which can ingest your source code, assemble it and then produce an application image which is ready to be deployed. In the case of the Oshinko source-to-image builders, an Apache Spark cluster will also be created dynamically for your application when it runs.
+
+[[setup]]
+== Setup your coding environment
+
+To utilize the source-to-image workflow you will need to have a https://git-scm.com[Git] repository to store your work. This repository is then used by the source-to-image builders to create the final application image which will be run on OpenShift. For the purposes of this tutorial your repository will need to be accessible by the OpenShift cluster, the easiest way to do this is by using an external service like https://github.com[GitHub] or https://bitbucket.com[BitBucket].
+
+Create a new repository on your preferred platform and make a local clone of it, if you need help with this process there are instructions available in https://confluence.atlassian.com/get-started-with-bitbucket/create-a-repository-861178559.html[BitBucket's documentation] and https://help.github.com/articles/create-a-repo/[GitHub's documentation].
+
+With your freshly minted repository you are ready to start coding!
+
+[[stack]]
+== Choose your language and technology stack
+
+This tutorial has been developed for multiple languages and HTTP frameworks. Functionally, all these microservices work in the same manner with similar results to an end user. In this way they represent a black box style microservice which could be replaced with another component that satisfies the same input and output requirements.
+
+Each of these language specific tutorials provide instructions on how to structure your repository and what files to create. You will also learn how each microservice works and the commands necessary to build and launch them in OpenShift. Because the source-to-image templates that you will use are language specific, you must choose a single implementation to exist in your repository. When you are finished, return here to learn about interacting with your microservice and exposing it outside of OpenShift. Please choose one of the following language and framework options:
+
+* link:/assets/my-first-radanalytics-app/my-first-radanalytics-app-python.html[Python with Flask]
```

Comment:
Within this link there is the tree command. 'tree' command is not found on Mac.

Reply:
Yeah, need homebrew. Is it worth it to add those instructions? Many Mac developers would have brew installed IMHO.

Reply:
i'm thinking maybe best to change this to an `ls` or something. i need to dig out a mac to experiment on.